orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/HttpClientUtils.java

```diff
@@ -0,0 +1,101 @@
+ */
+
+package com.netflix.spinnaker.orca.pipeline.util;
+
+import com.google.common.io.CharStreams;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.protocol.HttpCoreContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class HttpClientUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientUtils.class);
+  private static final int MAX_RETRIES = 5;
+  private static final int RETRY_INTERVAL = 3000;
+  private static final int TIMEOUT_MILLIS = 3000;
+  private static List<Integer> RETRYABLE_500_HTTP_STATUS_CODES =Arrays.asList(
+      HttpStatus.SC_SERVICE_UNAVAILABLE,
+      HttpStatus.SC_INTERNAL_SERVER_ERROR
+  );
+
+  private static CloseableHttpClient httpClient = httpClientWithServiceUnavailableRetryStrategy();
+
+  private static CloseableHttpClient httpClientWithServiceUnavailableRetryStrategy() {
+    HttpClientBuilder httpClientBuilder =  HttpClients.custom()
+      .setServiceUnavailableRetryStrategy(new ServiceUnavailableRetryStrategy() {
+        @Override
+        public boolean retryRequest(HttpResponse response, int executionCount, HttpContext context) {
+          int statusCode = response.getStatusLine().getStatusCode();
+          HttpUriRequest currentReq = (HttpUriRequest) context.getAttribute(
+            HttpCoreContext.HTTP_REQUEST);
+
+          LOGGER.info("Response code {} for {}", statusCode, currentReq.getURI());
+          boolean shouldRetry = (statusCode == 429 || RETRYABLE_500_HTTP_STATUS_CODES.contains(statusCode)) && executionCount <= MAX_RETRIES;
+          if (shouldRetry) {
+            LOGGER.error("Retrying request on response status {}. Count {} Max is {}", statusCode, executionCount, MAX_RETRIES);
+          }
+
+          return shouldRetry;
+        }
+
+        @Override
+        public long getRetryInterval() {
+          return RETRY_INTERVAL;
+        }
+      }
+    );
+
+    httpClientBuilder.setDefaultRequestConfig(
+      RequestConfig.custom()
+        .setConnectionRequestTimeout(TIMEOUT_MILLIS)
+        .setConnectTimeout(TIMEOUT_MILLIS)
+        .build()
+    );
+
+    return httpClientBuilder.build();
+  }
+
+  static String httpGetAsString(String url) throws IOException {
+    try (CloseableHttpClient client = httpClient) {
```

Comment:
this is going to call close on the client after the first fromUrl happens - since the client is static does that leave it in an unusable state for any other callers?